### PR TITLE
Fix admin users not connection correctly to ROS

### PR DESCRIPTION
**Pull request by cmelchior**
Depends on #4752 being merged first

Fixes admin users getting saved with the wrong token type. Should have used Normal instead of Admin, since Admin is only for the admin token, not admin users :/

### CODE REVIEW

File: realm/realm-library/src/objectServer/java/io/realm/SyncSession.java [hunk 1]
```diff
@@ -309,9 +310,11 @@ protected void onSuccess(AuthenticateResponse response) {
                             configuration.getPath(),
                             configuration.shouldDeleteRealmOnLogout()
                     );
-                    getUser().getSyncUser().addRealm(configuration.getServerUrl(), desc);
+                    URI realmUrl = configuration.getServerUrl();
+                    getUser().getSyncUser().addRealm(realmUrl, desc);
                     // schedule a token refresh before it expires
```
Comment (beeender) [inline realm/realm-library/src/objectServer/java/io/realm/SyncSession.java]: this comment should go to line 317 then.

File: realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/BaseIntegrationTest.java [hunk 2]
```diff
@@ -51,8 +56,29 @@ public static void tearDown () throws Exception {
         try {
             HttpUtils.stopSyncServer();
             RealmLog.setLevel(originalLogLevel);
+            deleteRosFiles();
         } catch (Exception e) {
             RealmLog.error("Failed to stop Sync Server", e);
         }
     }
+
+    // Cleanup filesystem to make sure nothing lives for the next text.
```
Comment (nhachicha) [inline realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/BaseIntegrationTest.java]: next test

File: realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ManagementRealmTests.java [hunk 3]
```diff
@@ -51,7 +53,42 @@
...
+        looperThread.closeAfterTest(realm);
+        looperThread.runAfterTest(new Runnable() {
+            @Override
+            public void run() {
+                user.logout();
+            }
+        });
+        realm.beginTransaction();
+        // Invalid Permission offer
+        realm.copyToRealm(new PermissionOffer("*", true, true, false, null));
```
Comment (nhachicha) [inline realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ManagementRealmTests.java]: it's outside the scope of this PR, but is it possible to refactor `PermissionOffer` to use a builder pattern? reading three boolean, can't know which permission they affect

File: realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ManagementRealmTests.java [hunk 4]
```diff
@@ -51,7 +53,42 @@
     @Rule
     public RunInLooperThread looperThread = new RunInLooperThread();
 
-    @Ignore("TODO: Test is currently flaky. See https://github.com/realm/realm-java/pull/4066")
+    // This is primarily a test making sure that an admin user actually connects correctly to ROS.
```
Comment (nhachicha) [inline realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ManagementRealmTests.java]: are you considering the fact that you can send a Permission & get an error code a "correct connection to ROS"?

Comment (cmelchior) [inline realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ManagementRealmTests.java]: Yes, I'm using the fact that the statusCode/statusMessage is written by ROS.

File: realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ManagementRealmTests.java (hunk 3 above)
Comment (cmelchior) [inline realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ManagementRealmTests.java]: Yes, I'm doing that as part of #4558

**Comment by cmelchior** [comment]
Replaced by #4760

**Pull request closed without merging**